### ISSUE TITLE
Author information lost.

### ISSUE THREAD

**Issue by netroby** [kubernetes/test-infra#2479]
A normal commit merge log will looks like 
```
commit aaf4682171f1ffabd3673b82cb71fdc3d8e5317e
Author: Meir Fischer <meirfischer@gmail.com>
Commit: Brad Fitzpatrick <bradfitz@golang.org>

    net/http/fcgi: expose cgi env vars in request context
    
    The current interface can't access all environment
    variables directly or via cgi.RequestFromMap, which
    only reads variables on its "white list" to be set on
    the http.Request it returns. If an fcgi variable is
    not on the "white list" - e.g. REMOTE_USER - the old
    code has no access to its value.
    
    This passes variables in the Request context that aren't
    used to add data to the Request itself and adds a method
    that parses those env vars from the Request's context.
    
    Fixes #16546
    
    Change-Id: Ibf933a768b677ece1bb93d7bf99a14cef36ec671
    Reviewed-on: https://go-review.googlesource.com/40012
    Run-TryBot: Brad Fitzpatrick <bradfitz@golang.org>
    TryBot-Result: Gobot Gobot <gobot@golang.org>
    Reviewed-by: Brad Fitzpatrick <bradfitz@golang.org>
```

The Author: section contains the origin PR author.

And the k8s-bot just made author information lost

that's not good.
```
commit 7f0004de5acd1b5dbb6e91f04908e05444bf86aa
Merge: e4752d612a c9535869fc
Author: Kubernetes Submit Queue <k8s-merge-robot@users.noreply.github.com>
Commit: GitHub <noreply@github.com>

    Merge pull request #44379 from guangyouyu/garyyu_master
    
    Automatic merge from submit-queue
    
    Bugfix: cloud-controller-manager routeController.run() block in WaitForCacheSync
```

**Comment by eparis** [comment]
That's a merge commit. That is correct. The bot was responsible for the merge commit. Start by looking at `Merge: e4752d612a c9535869fc` It merged the SHAs on the left (e4752d612a) and the right (c9535869fc) as this new commit. If you are interested in the commits inside the left and the right branch you'd have to `git log` on those. Since the left side is usually the HEAD the right side is more interesting. In this case:
```
$ git show c9535869fc
commit c9535869fcf4677bdde181c19159aa426dafbbbc
Author: garyyu(于广游) <garyyu@tencent.com>
Date:   Wed Apr 12 13:03:41 2017 +0800
```
Which is the original commit. Everything is working like it is supposed to work here...

**Comment by rmmh** [comment]
This is exactly what happens when you press the "Merge" button on Github, since that's the interface that the merge bot is using. That's why it's "Commit: GitHub <noreply@github.com>".
